rohmu/object_storage/azure.py

```diff
@@ -156,36 +162,46 @@ def copy_file(self, *, source_key, destination_key, metadata=None, **kwargs):
                     )
                 )
 
-    def get_metadata_for_key(self, key):
+    def get_metadata_for_key(self, key: str) -> Metadata:
         path = self.format_key_for_backend(key, remove_slash_prefix=True, trailing_slash=False)
         items = list(self._iter_key(path=path, with_metadata=True, deep=False))
         if not items:
             raise FileNotFoundFromStorageError(path)
         expected_name = path.rsplit("/", 1)[-1]
+        item: Optional[IterKeyItem]
         for item in items:
+            assert item is not None  # for mypy
             # We expect single result but Azure listing is prefix match so we need to explicitly
             # look up the matching result
+            item_name: str
             if item.type == KEY_TYPE_OBJECT:
+                assert isinstance(item.value, dict)
                 item_name = item.value["name"]
             else:
+                assert isinstance(item.value, str)
                 item_name = item.value
             if item_name.rstrip("/").rsplit("/", 1)[-1] == expected_name:
                 break
         else:
             item = None
         if not item or item.type != KEY_TYPE_OBJECT:
             raise FileNotFoundFromStorageError(path)  # not found or prefix
```

Comment:
nit perhaps, but this is more readable for me at least:
```suggestion
        for item in items:
            # We expect single result but Azure listing is prefix match so we need to explicitly
            # look up the matching result
            item_name: str
            if item.type == KEY_TYPE_OBJECT:
                assert isinstance(item.value, dict)
                item_name = item.value["name"]
            else:
                assert isinstance(item.value, str)
                item_name = item.value
            if item_name.rstrip("/").rsplit("/", 1)[-1] == expected_name:
                break
        else:
            raise FileNotFoundFromStorageError(path)
        if item.type != KEY_TYPE_OBJECT:
            raise FileNotFoundFromStorageError(path)
```